.github/workflows/main.yaml

```diff
@@ -0,0 +1,135 @@
+name: main
+
+on:
+  push:
+    branches:
+      - "**"  
+  pull_request:
+    branches:
+      - "**"
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.10.9
+      - name: install dependencies
+        run: pip install .
+      - name: lint
+        uses: wearerequired/lint-action@v2
+        with:
+          continue_on_error: false
+          black: true
+          flake8: true
+  bandit:
+    name: bandit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.10.9
+      - name: install dependencies
+        run: pip install .
+      - name: bandit
+        run: bandit -r -c pyproject.toml .
+  cognitive:
+    name: cognitive
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.10.9
+      - name: install dependencies
+        run: pip install .
+      - name: cognitive
+        run: flake8 . --max-cognitive-complexity=5
+  isort:
+    name: isort
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.10.9
+      - name: install dependencies
+        run: pip install .
+      - name: isort
+        uses: isort/isort-action@v1.1.0
+  pydocstyle:
+    name: pydocstyle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.10.9
+      - name: install dependencies
+        run: pip install .
+      - name: pydocstyle
+        run: pydocstyle .
+  test:
+    name: test
+    runs-on: ubuntu-latest
+    needs: [bandit, cognitive, isort, lint, pydocstyle]
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.10.9
```

Comment:
I've done this for the Python versions, since it's quite clear what we support. 

I would prefer to wait until we have more users to discover which OSs they are actually using before making a decision about the OSs to test.

Reply:
[603f99d](https://github.com/opencosmos/datacosmos-sdk/pull/1/commits/603f99da25097a670a3f001b25e4e782ac7cac7d)

Reply:
I am not sure what the adoption rate is like for new python versions but a minimum of 3.10 still seems fairly high. I would recommend at least supporting version 3.8, for example it is the default version on Ubuntu 18/20 as a reference.

Reply:
I chose 3.10 as the minimum (this is also stated in the `pyproject.toml`) so we have the benefit of `switch` statements.

Here's the quickest thing I found about Python version adoption: https://git.afpy.org/mdk/python-versions

Reply:
I see, doing if-elses in Python is not far from the match-case statements introduced in 3.10 in terms of functionality/LoC. Based on the link you've added Python versions 3.10 and later only accounts for < 10% of the user base and even suggests we should support version 3.7 as the earliest. However 3.7 is about to hit EoL this year so likely we can jump ahead to 3.8 prematurely.

Reply:
bf2aaa9 and 8955200